index.ts

```diff
@@ -12,21 +12,20 @@ function getLangCode(tag: string) {
   return tag.substring(0, 2);
 }
 
-export default function gettextFactory(langTable: LanguageTable, currentLocale = defaultLocale) {
-  function getMessages(locale: string): MessageTable {
-    return langTable[locale] || {};
-  }
-
+export default function gettextFactory(
+  langTable: LanguageTable,
+  currentLocale = defaultLocale,
+  fallbackLocale = defaultLocale,
+) {
   // Return a named function so it shows up in stack traces.
   return function gettext(msgid: unknown): string {
     const msgidString = String(msgid);
 
-    return (
-      getMessages(currentLocale)[msgidString] ||
-      getMessages(getLangCode(currentLocale))[msgidString] ||
-      getMessages(defaultLocale)[msgidString] ||
-      getMessages(getLangCode(defaultLocale))[msgidString] ||
-      msgidString
-    );
+    const msgTable = langTable[currentLocale] ||
```

Comment:
Hoist defining `msgTable` outside the returned function. It's a constant that does not need to be reevaluated in each `gettext` call, and it's wasteful to close over the full `langTable` when only one member is needed.

```ts
export default function gettextFactory(...) {
  const msgTable = langTable[currentLocale] || ...;

  return function gettext(msgid: unknown): string {
    const msgidString = String(msgid);
    return msgTable[msgidString] || msgidString;
  };
}
```